osu.Game/OsuGame.cs

```diff
@@ -346,11 +344,24 @@ public void PresentBeatmap(BeatmapSetInfo beatmap, Predicate<BeatmapInfo> diffic
                     return;
                 }
 
-                // Find first beatmap that matches our predicate.
-                var first = databasedSet.Beatmaps.Find(difficultyCriteria) ?? databasedSet.Beatmaps.First();
+                // Find beatmaps that match our predicate.
+                var beatmaps = databasedSet.Beatmaps.Where(b => difficultyCriteria?.Invoke(b) ?? true);
+                if (!beatmaps.Any())
+                    beatmaps = databasedSet.Beatmaps;
+
+                var selection = DifficultyRecommender.GetRecommendedBeatmap(beatmaps);
+
+                // fallback if a difficulty can't be recommended, maybe we are offline
+                if (selection == null)
+                {
+                    if (difficultyCriteria != null)
+                        selection = beatmaps.First();
+                    else
+                        selection = databasedSet.Beatmaps.Find(b => b.Ruleset.Equals(Ruleset.Value)) ?? databasedSet.Beatmaps.First();
```

Comment:
Not sure this discarding `difficultyCriteria` to only check ruleset is correct. Sounds like it might break some previous changes in this area.

Reply:
the first path of the if statement would use the beatmaps filtered by ``difficultyCriteria``